src/Microsoft.VisualStudio.Services.Agent/Util/IOUtil.cs

```diff
@@ -151,6 +151,12 @@ public static string GetWorkPath(IHostContext hostContext)
                 settings.WorkFolder);
         }
 
+        public static string GetHomePath()
+        {
+            var envHome = Constants.Agent.Platform == Constants.OSPlatform.Windows ? "HOMEPATH" : "HOME";
```

Comment:
...i could be wrong, but that was my fuzzy interpretation based on how long i looked into it.